lib/index.js

```diff
@@ -808,6 +810,58 @@ GA.prototype.productClickedEnhanced = function(track) {
 };
 
 /**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListViewedEnhanced = function(track) {
+  var props = track.properties(); 
+  var products = track.products();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
+    window.ga('ec:addImpression', {
```

Comment:
We actually support more values here (update to our specification just recently occurring). Going to add these here, but the rest look golden!

id: product.product_id || product.sku,
name: product.name,
category: product.category || track.category(),
list: props.list_id || track.category() || 'products',
brand: products.brand,
variant: products.variant,
price: products.price,
position: products.map(function(x) { return x.product_id; }).indexOf(product.product_id)